0_From_Code_First_To_First_Principles/1_code_first/assignment/src/modules/user/commands/use-cases/create-user/create-user.controller.ts

```diff
@@ -0,0 +1,57 @@
+import { plainToClass } from "class-transformer";
+import { validate } from "class-validator";
+import { FailResponseDto } from "../../../../../libs/api/fail.response.dto";
+import { ICreateUserController } from "./contracts/create-user.controller.interface";
+import { CreateUserRequestDto } from "./contracts/create-user.request-dto";
+import { CreateUserResponseDto } from "./contracts/create-user.response-dto";
+import { ICreateUserUseCase } from "./contracts/create-user.use-case.interface";
+
+export class CreateUserController implements ICreateUserController {
+  constructor(private readonly useCase: ICreateUserUseCase) {}
+
+  async invoke(message: any) {
```

Comment:
Methods/functions shouldn’t be longer than 20 lines (and ideally much less)

Reply:
Good one! I had a lot of fun refactoring this one.  What do you think of the result?